File: CS50/Week2/Greedy.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
     float owed;
     int coins = 0;
     int quart=0, dime=0, nickles=0, penny=0;
    do{
    printf("O hai! How Much change is owed?");
    owed = get_float();
  }while (owed < 0);

    //convert from float to integer
  while (owed > 0) {
    while (owed >= 25 ){
        coins ++
        quart ++
        owed = owed - 25;
      }
    while (owed >= 10){
        coins ++
        dime ++
          owed = owed -10;
        }
    while (owed >= 5)
    {
      coins ++
      nickles ++
      owed = owed -5
      }
    while (owed >= 1)
    {
    coins ++
    penny ++
      owed = owed -1
    }
  }
    //print number of coins used

    printf("%i\n", coins);
}

File: CS50/Week6/HW/greedy.py
owed = float(input("Hello how much change do I owe: "))

coins = int(0)
quart = int(0)
dime = int(0)
nickles = int(0)
penny = int(0)

while (owed > 0):
    while(owed >= .25):
        coins += 1
        quart += 1
        owed = owed - .25
    while(owed >= .10):
        coins += 1
        dime += 1
        owed = owed - .10
    while(owed >= .5):
        coins += 1
        nickles += 1
        owed = owed - .5
    while(owed >= .1):
        coins += 1
        penny += 1
        owed = owed - .01

print(coins)